src/main/java/com/teragrep/lsh_01/BasicAuthentication.java

```diff
@@ -0,0 +1,68 @@
+/*
+  logstash-http-input to syslog bridge
+  Copyright 2024 Suomen Kanuuna Oy
+
+  Derivative Work of Elasticsearch
+  Copyright 2012-2015 Elasticsearch
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package com.teragrep.lsh_01;
+
+import com.teragrep.jai_02.CredentialLookup;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.Base64;
+
+public class BasicAuthentication {
+
+    Base64.Decoder decoder = Base64.getDecoder();
+    CredentialLookup credentialLookup;
+
+    public BasicAuthentication() {
+        BufferedReader br;
```

Comment:
create BasicAuthenticationFactory to create this, so there is no code in the costructor but the Factory has create method that returns BasicAuthentication